src/Parser/Recursion.php

```diff
@@ -31,11 +50,27 @@ function isArrayRecursion(array &$var) : bool
         {
             if ($known === $var)
             {
-                return true;
+                // test if they are copies of one another,
+                // or they are reference to the same array
+                $var[ $this->marker ] = 1;
+                $recursion = !empty($known[ $this->marker ]);
+                unset($var[ $this->marker ]);
+                return $recursion;
```

Comment:
This will still return if `$known === $var` so it will provide false *negatives* now (Which is worse than false positives!)

Reply:
@jnvsor nice catch, sorry about that

Reply:
Changed it in 772415a and now the return is triggered only when there is a recursion detected